### ISSUE TITLE
Proposal: Not remove MaterialModule in next release

### ISSUE THREAD

**Issue by karolmie1** [angular/material2#4261]
#### Bug, feature request, or proposal: Proposal

Just today I've replaced all MaterialModules with more specific module in my whole application, for example: 

```
-import { MaterialModule } from '@angular/material';
 
+import { MdCardModule, MdButtonModule } from '@angular/material';
+const MATERIAL_MODULE = [ MdCardModule, MdButtonModule ];
```

#### What is the expected behavior?
For a medium sized application I expected some decrease in application size as I don't use all modules.
Before: https://ibb.co/eaw0wQ
After: https://ibb.co/fcR0wQ

#### What is the current behavior?
The application size did almost not change - really insignificant decrease, few kb. I guess that with es5 modules UglifyJS does good enough job for the change to have any impact. Or maybe even direct imports is not enough to weed out unnecessary parts of code.

Maybe it's just my specific setup (angular2-webpack-starter type). Did you experience any difference after switching to more specific imports?

#### What are the steps to reproduce?

If anyone interested in replicating the result: 
https://bitbucket.org/kmierzej/fragaria-webapp/commits/6c78b86731c4e59d2a0527ed7ccdbe5737f72563

**Comment by DevVersion** [comment]
This should be fixed with https://github.com/angular/material2/commit/76faae5f378f2949bbe215df38369190773eba60.

Can you try installing `npm i angular/material2-builds`?

**Comment by karolmie1** [comment]
I did, but unfortunately the results did not vary. Probably webpack fault. There is a huge discussion about this pure annotation thing.
I see the point of this depreciation now thought. I would leave this issue for a while, for someone else to maybe share their results. But that is up to you - just close if you feel like this.

**Comment by jelbourn** [comment]
There's another issue tracked in #4137 about webpack not being able to do this. We're working on something now to fix this; MaterialModule is still going to go away, but only once the benefit applies without a lot of extra work.
